lib/awilixManager.ts

```diff
@@ -0,0 +1,96 @@
+import type { AwilixContainer } from 'awilix'
+
+declare module 'awilix' {
+  // @ts-ignore
+  interface ResolverOptions<T> {
+    asyncInit?: boolean | string
+    asyncInitPriority?: number // lower means it gets initted earlier
+    asyncDispose?: boolean | string
+    asyncDisposePriority?: number // lower means it gets disposed earlier
+    eagerInject?: boolean
+  }
+}
+
+export type AwilixManagerConfig = {
+  diContainer: AwilixContainer
+  asyncInit?: boolean
+  eagerInject?: boolean
+}
+
+export class AwilixManager {
+  private readonly config: AwilixManagerConfig
+
+  constructor(config: AwilixManagerConfig) {
+    this.config = config
+  }
+
+  async executeInit() {
+    if (this.config.eagerInject) {
+      eagerInject(this.config.diContainer)
+    }
+
+    if (this.config.asyncInit) {
+      await asyncInit(this.config.diContainer)
+    }
+  }
+
+  async executeDispose() {
+    await asyncDispose(this.config.diContainer)
+  }
+}
+
+export async function asyncInit(diContainer: AwilixContainer) {
+  const dependenciesWithAsyncInit = Object.entries(diContainer.registrations)
+    .filter((entry) => {
+      return entry[1].asyncInit
+    })
+    .sort((entry1, entry2) => {
+      const asyncInitPriority1 = entry1[1].asyncInitPriority ?? 1
+      const asyncInitPriority2 = entry2[1].asyncInitPriority ?? 1
+
+      return asyncInitPriority1 - asyncInitPriority2
+    })
```

Comment:
I think it would be a good idea to add a tiebreaker here and in the `asyncDispose` sort function in case of equal priority, just to keep initting and disposing deterministic. When there is a tie, break it using the dependency's name. 

Off the top of my head it might look something like this, or maybe something more explicit and less clever with if statements if desired.

```suggestion
    .sort((entry1, entry2) => {
      const [key1, resolver1] = entry1
      const [key2, resolver2] = entry2
      const asyncInitPriority1 = resolver1.asyncInitPriority ?? 1
      const asyncInitPriority2 = resolver2.asyncInitPriority ?? 1

      return (10 * (asyncInitPriority1 - asyncInitPriority2)) + (key1.localeCompare(key2))
    })
```

Reply:
great idea, thanks!